### ISSUE TITLE
BezierCurve.interpolate() returns NaN under some situations

### ISSUE THREAD

**Issue by mjschlot** [BabylonJS/Babylon.js#2628]
For some of the animations I am making it I want to use the following BezierCurveEase (which is the same as I often use in CSS animations):
`new BABYLON.BezierCurveEase(0, 0, 0, 1);`

The animation does not work though. When I dug in deeper I found that the underlying method of `BezierCurve.interpolate()` is returning `NaN` when `t` is zero in the animation (so first frame).

I confirmed the above with:
`let test = BABYLON.BezierCurve.interpolate(0, 0, 0.4, 0, 0);
console.log(test);`

I am not familiar enough with the math involved to know how to properly fixed. But I did a quick debug and stepped through it. The issue happens on this line:
`var slope = 1.0 / (3.0 * f0 * refinedT2 + 2.0 * f1 * refinedT + f2);`

The above is equivalent to `1.0 / 0` when `t` (first argument of `BABYLON.BezierCurve.interpolate`) is 0.

How should this be addressed? Should we add a condition to the calculation of slope to set slope to 0 when the delimiter is 0? Or maybe we simply add an if statement of `t === 0` at the top of the `BezierCurve.interpolate()` and return 0 immediately?

**Comment by deltakosh** [comment]
I'm more for the latter:)

________________________________

**Comment by mjschlot** [comment]
I am too. .. Just to confirm `t` of 0 would always return 0 in a bezier curve method? I can't think of a situation where it wouldn't. If that is correct, wouldn't the same situation be true when `t` is 1. Would it make sense to always return 1 if `t` is 1?

**Comment by deltakosh** [comment]
we should return 0 only if t = x1 = x2 = 0 only actually, right? t is not really the problem

**Comment by deltakosh** [comment]
So I would suggest to test (3.0 * f0 * refinedT2 + 2.0 * f1 * refinedT + f2) and just make sure it is not 0 else we could just set slope to number.MAXVALUE ;)

**Comment by deltakosh** [comment]
Up?

**Comment by mjschlot** [comment]
Hi David. Other projects have been keeping me very busy lately. I'll submit a PR on this and I think one or two other items late this week or this coming weekend.

**Comment by deltakosh** [comment]
sure no problem..take your time..Just wanted to get a sense of the timing ;)

**Comment by mjschlot** [comment]
No need to respond, but I wanted to provide a quick update: There is no longer an error when animating. The `BezierCurve.interpolate` function still returns `NaN` though.

Also `NaN` is is not being returned because of dividing by zero like I thought. Chrome used to return `NaN` when dividing by zero but now returns `Infinity` which is to spec and consistent with other browsers (I don't know when this change happened in Chrome; may be recent or may have happened years ago).

`NaN` is returned for another reason so the fix we discussed is not the correct answer because either Chrome recently changed or my earlier investigation was wrong that dividing by zero was the cause. I will look into this more and implement a good solution to `BezierCurve.interpolate`, but there has been a change somewhere else in the code where `NaN` does not appear to be causing an error when the easing function returns that value. I started hunting down why animations no longer fail when `NaN` is the new value during animation, but ran out of time before I found the change that addressed that. And at this point I don't think it matters why it no longer errors when animating, but fixing this bug in `BezierCurve.interpolate` is still a good idea which I will look at again this week and submit a PR for.
